File: src/domain/entities/product.ts
import type { Inventory } from "./inventory";
import { Order } from "./order";
import type { Cart } from "./cart";
import type { SubCategory } from "./sub_category";

export interface Product {
	id_product: number;
	id_seller: number;
	id_inventory: number;
	id_category: number;
	id_sub_category: number;
	product_name: string;
	description: string;
	price: number;
	stock: number;
	image_url: string;
	created_at: Date;
	inventory: Inventory;
	subcategory?: SubCategory;
	order?: Order[];
	cart?: Cart[];
}

File: backend/realtime_service/internal/models/product.go
package models

// Product representa un producto simplificado para eventos en tiempo real.
// Sincronizado con ProductEntity del rest_service.
type Product struct {
	IDProduct     int     `json:"id_product"`
	IDSeller      int     `json:"id_seller"`
	IDInventory   int     `json:"id_inventory"`
	IDCategory    int     `json:"id_category"`
	IDSubCategory int     `json:"id_sub_category"`
	ProductName   string  `json:"product_name"`
	Description   string  `json:"description,omitempty"`
	Price         float64 `json:"price"`
	Stock         int     `json:"stock"`
	ImageURL      string  `json:"image_url,omitempty"`
}
